htmresearch/frameworks/nlp/classification_model.py

```diff
@@ -19,335 +19,183 @@
 # http://numenta.org/licenses/
 # ----------------------------------------------------------------------
 
-from collections import defaultdict, OrderedDict
-import copy
 import cPickle as pkl
 import numpy
-import operator
 import os
-import random
 import shutil
 
-import simplejson as json
-
 from htmresearch.support.text_preprocess import TextPreprocess
 
 
-
 class ClassificationModel(object):
   """
   Base class for NLP models of classification tasks. When inheriting from this
   class please take note of which methods MUST be overridden, as documented
   below.
   """
-  # TODO: use nupic.bindings.math import Random
 
   def __init__(self,
                verbosity=1,
                numLabels=3,
-               modelDir="ClassificationModel"):
+               filterText=False,
+               **kwargs):
     """
-    If there are no labels set numLabels=0.
+    @param verbosity  (int) Verbosity level. Larger values lead to more
+                            printouts. 0 implies nothing will be printed.
+
+    @param numLabels  (int) The maximum number of categories.
```

Comment:
Previously `numLabels` was used as the max number of categories a model can infer. The use in this refactored model, however, implies `numLabels` is the total number of categories in the dataset. We should be explicit about the definition here.